NitroxClient/GameLogic/EscapePodManager.cs

```diff
@@ -1,77 +1,67 @@
 ﻿using System.Collections.Generic;
 using NitroxClient.Communication.Abstract;
-using NitroxClient.Communication.Packets.Processors.Abstract;
 using NitroxClient.GameLogic.Helper;
 using NitroxClient.Unity.Helper;
 using NitroxModel.DataStructures.GameLogic;
 using NitroxModel.Logger;
 using NitroxModel.Packets;
 using UnityEngine;
 
-namespace NitroxClient.Communication.Packets.Processors
+namespace NitroxClient.GameLogic
 {
-    public class BroadcastEscapePodsProcessor : ClientPacketProcessor<BroadcastEscapePods>
+    public class EscapePodManager
     {
-        public static bool SURPRESS_ESCAPE_POD_AWAKE_METHOD;
-
         private readonly IPacketSender packetSender;
-        private IMultiplayerSession multiplayerSession;
+        private readonly IMultiplayerSession multiplayerSession;
+
+        public static bool SURPRESS_ESCAPE_POD_AWAKE_METHOD;
         private readonly Vector3 playerSpawnRelativeToEscapePodPosition = new Vector3(0.9f, 2.1f, 0);
         private readonly Dictionary<string, GameObject> escapePodsByGuid = new Dictionary<string, GameObject>();
 
-        private string myEscapePodGuid;
+        public string MyEscapePodGuid;
 
-        public BroadcastEscapePodsProcessor(IPacketSender packetSender, IMultiplayerSession multiplayerSession)
+        public EscapePodManager(IPacketSender packetSender, IMultiplayerSession multiplayerSession)
         {
             this.packetSender = packetSender;
             this.multiplayerSession = multiplayerSession;
         }
 
-        public override void Process(BroadcastEscapePods packet)
+        public void AssignPlayerToEscapePod(List<EscapePodModel> escapePods, string assignedEscapePodGuid)
```

Comment:
Let's change to `AssignPlayerToEscapePod(EscapePodModel escapePod)` and do the `EscapePodModel escapePod = escapePods.Find(x => x.Guid == assignedEscapePodGuid);` right before calling.